### PR TITLE
Prompt users to log in when they click the "Buy" button

### PR DESCRIPTION
**Pull request by sweir27**
This PR addresses: https://artsyproduct.atlassian.net/browse/PURCHASE-487

It requires that users log in before they are able to create an order.

login-modal

### CODE REVIEW

File: src/desktop/config.coffee [hunk 1]
```diff
@@ -81,7 +81,7 @@ module.exports =
   MOBILE_MEDIA_QUERY: "only screen and (max-width: 640px)"
   MOBILE_URL: 'https://m.artsy.net'
   NEW_ARTIST_PAGE_THRESHOLD: 10 # all users will get the new page.
-  NEW_AUTH_MODAL: false
+  NEW_AUTH_MODAL: true
```
Comment (sweir27) [inline src/desktop/config.coffee]: @l2succes is this okay? I updated this config to default to `true` so people don't inadvertently see the old modals locally

File: src/mobile/config.coffee [hunk 2]
```diff
@@ -35,7 +35,7 @@ module.exports =
   MAILCHIMP_AUCTION_LIST_ID: 'b7b9959ee0'
   MAILCHIMP_KEY: null
   MOBILE_MARKETING_SIGNUP_MODALS: '[{"slug":"ca1","copy":"An art collection for every budget","image":"http://files.artsy.net/images/modal-collect-art.jpg"},{"slug":"ca2","copy":"Buy art from the best galleries and auction houses","image":"http://files.artsy.net/images/modal-collect-art.jpg"},{"slug":"ca3","copy":"Discover and Buy Works from Seattle Art Fair 2017","image":"http://files.artsy.net/images/seattle-art-fair-modal.jpg","photoCredit":"Sarah Cain, waves, 2016; Courtesy of the artist and Galerie Lelong & Co., New York"}]'
-  MOBILE_NEW_AUTH_MODAL: false
+  MOBILE_NEW_AUTH_MODAL: true
```
Comment (sweir27) [inline src/mobile/config.coffee]: (same here for mobile)

File: src/desktop/config.coffee (hunk 1 above)
Comment (l2succes) [reply src/desktop/config.coffee]: Yup that's fine, we'll be removing that flag soon when we clean up the legacy /auth app code

File: src/mobile/config.coffee (hunk 2 above)
Comment (l2succes) [reply src/mobile/config.coffee]: ditto!

**Comment by ashfurrow** [comment]
Cool, thanks Luc!

**Pull request merged**